python/ray/train/_internal/stats.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import threading
+from typing import Dict, Optional
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.schema import TrainDatasetInfo, TrainRunInfo, TrainWorkerInfo
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote(num_cpus=0)
+class TrainStatsActor:
+    def __init__(self):
+        self.train_runs = dict()
+
+    def register_train_run(self, run_info: TrainRunInfo):
+        # Register a new train run.
+        self.train_runs[run_info.id] = run_info
+
+    def get_train_run(self, run_id: str) -> Optional[TrainRunInfo]:
+        # Retrieve a registered run with its id
+        return self.train_runs.get(run_id, None)
+
+    def get_all_train_runs(self) -> Dict[str, TrainRunInfo]:
+        # Retrieve all registered train runs
+        return self.train_runs
+
+
+TRAIN_STATS_ACTOR_NAME = "train_stats_actor"
+TRAIN_STATS_ACTOR_NAMESPACE = "_train_stats_actor"
+
+_stats_actor_lock: threading.RLock = threading.RLock()
+
+
+def get_or_launch_stats_actor():
+    """Create or launch a `TrainStatsActor` on the head node."""
+    with _stats_actor_lock:
+        return TrainStatsActor.options(
+            name=TRAIN_STATS_ACTOR_NAME,
+            namespace=TRAIN_STATS_ACTOR_NAMESPACE,
+            get_if_exists=True,
+            lifetime="detached",
+            resources={"node:__internal_head__": 0.001},
+        ).remote()
+
+
+class TrainRunStatsManager:
+    """A class that aggregates and reports train run info to TrainStatsActor.
+
+    This manager class is created on the train controller layer for each run.
+    """
+
+    def __init__(self) -> None:
+        self.stats_actor = get_or_launch_stats_actor()
+
+    def register_train_run(
+        self,
+        run_id: str,
+        run_name: str,
+        trial_name: str,
```

Comment:
`run_id` and `run_name` should be enough for a Train Run. In the future we may consider remove `trial_name` after we separate Train and Tune.

Reply:
This may help differentiate if a user launches multiple runs with the same experiment name.

we can probably remove it and replace with a timestamp instead.

Reply:
The user can just differentiate these with `run_id`. This is kind of a tune dependency that I don't want the dashboard to have.